includes/admin/meta-boxes/class-wc-meta-box-order-actions.php

```diff
@@ -28,18 +26,13 @@ public static function output( $post ) {
 		global $theorder;
 
 		// This is used by some callbacks attached to hooks such as woocommerce_order_actions which rely on the global to determine if actions should be displayed for certain orders.
+		// Avoid using this global with the `woocommerce_order_actions` filter, instead use the $order filter arg.
 		if ( ! is_object( $theorder ) ) {
 			$theorder = wc_get_order( $post->ID );
 		}
 
-		$order_actions = apply_filters(
-			'woocommerce_order_actions',
-			array(
-				'send_order_details'              => __( 'Email invoice / order details to customer', 'woocommerce' ),
-				'send_order_details_admin'        => __( 'Resend new order notification', 'woocommerce' ),
-				'regenerate_download_permissions' => __( 'Regenerate download permissions', 'woocommerce' ),
-			)
-		);
+		$theorder = $theorder instanceof WC_Order ? $theorder : null;
+		$order_actions = self::get_available_order_actions_for_order( $theorder );
```

Comment:
Hi @danielbitzer - Thanks for the PR! So we're trying not to add any new methods unless it is absolutely necessary. And for those cases, we also utilize the internal src namespace to do so. That way it is not publicly exposed. But in this case, I don't think we need a new method for this. Perhaps just put the docblock above the filter for clarity.

Reply:
HI @roykho! Would making the method private also be acceptable? That way it's not publicly exposed?

Reply:
Yes that would be better.